lua/telescope/actions/set.lua

```diff
@@ -50,71 +49,65 @@ set.edit = function(prompt_bufnr, command)
   if not entry then
     print("[telescope] Nothing currently selected")
     return
-  else
-    local filename, row, col
-    if entry.filename then
-      filename = entry.path or entry.filename
-
-      -- TODO: Check for off-by-one
-      row = entry.row or entry.lnum
-      col = entry.col
-    elseif not entry.bufnr then
-      -- TODO: Might want to remove this and force people
-      -- to put stuff into `filename`
-      local value = entry.value
-      if not value then
-        print("Could not do anything with blank line...")
-        return
-      end
-
-      if type(value) == "table" then
-        value = entry.display
-      end
-
-      local sections = vim.split(value, ":")
+  end
 
-      filename = sections[1]
-      row = tonumber(sections[2])
-      col = tonumber(sections[3])
+  local filename, row, col
+
+  if entry.filename then
+    filename = entry.path or entry.filename
+
+    -- TODO: Check for off-by-one
+    row = entry.row or entry.lnum
+    col = entry.col
+  elseif not entry.bufnr then
+    -- TODO: Might want to remove this and force people
+    -- to put stuff into `filename`
+    local value = entry.value
+    if not value then
+      print("Could not do anything with blank line...")
+      return
     end
 
-    local preview_win = state.get_status(prompt_bufnr).preview_win
-    if preview_win then
-      a.nvim_win_set_config(preview_win, {style = ''})
```

Comment:
I think we had this because sometimes preview options were applied to the main window. I am not sure if this would happen again if we remove that part.